src/tools/launcher/util/launcher_util.cc

```diff
@@ -170,9 +170,89 @@ std::wstring BashEscapeArg(const std::wstring& arg) {
   return GetEscapedArgument(arg, /* escape_backslash */ true);
 }
 
+std::wstring WindowsEscapeArg2(const std::wstring& s) {
```

Comment:
Where does this algorithm come from? Can you add a link here?

Reply:
Done.

https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/ describes all details.

I did not base my algorithm on that, but luckily they actually do the same.

Reply:
Thanks!!